src/analysis/convex/combination.lean

```diff
@@ -121,6 +122,23 @@ lemma finset.center_mass_filter_ne_zero :
 finset.center_mass_subset z (filter_subset _ _) $ λ i hit hit',
   by simpa only [hit, mem_filter, true_and, ne.def, not_not] using hit'
 
+namespace finset
+
+lemma center_mass_le_sup {s : finset ι} (hs : s.nonempty) {w : ι → R} (hw₀ : ∀ i ∈ s, 0 ≤ w i)
```

Comment:
can't you remove `hs` from the assumptions, and derive it from `hw₁` instead? Same thing in the next lemma.

Reply:
Bien vu !